### DISCUSSION TITLE
[Enhancement] Profile iOS and Android memory leaks

### DISCUSSION THREAD

**Discussion by imuller** [dotnet/maui#22925] [Q&A]
## Summary
Profile iOS and Android apps to find memory leaks. This will result in better apps.

## API Changes
When developing Xamarin.Forms, the only way to profile memory leaks is by using the Xamarin Profiler. And therefore its required to have a Visual Studio Enterprise license which costs $250 per month.

To help developers find these memory leaks they need the best tools. This also benefits the development with .NET MAUI. It’s a requested feature https://developercommunity.visualstudio.com/t/make-xamarin-profiler-available-for-professional-s/351884

Would .NET MAUI work with the platform specific profiles, like xCode Instruments or does it has its own profiler?

**Comment by jsuarezruiz** [comment]
cc @maddymontaquila

**Comment by jfversluis** [comment]
> Would .NET MAUI work with the platform specific profiles, like xCode Instruments or does it has its own profiler?

At the end of the line we still produce a iOS or Android or Windows app, so the platform tooling definitely should be able to run on it although it will probably cost a bit of effort to set it up as it's not integrated as a solution we might have.

**Comment by ghost** [comment]
We've moved this issue to the Backlog milestone. This means that it is not going to be worked on for the coming release. We will reassess the backlog following the current release and consider this item at that time. To learn more about our issue management process and to have better expectation regarding different types of issues you can read our [Triage Process](https://github.com/dotnet/maui/blob/main/docs/TriageProcess.md).

**Comment by AdamEssenmacher** [comment]
I'm working on a toolkit that has an attached property that can be used to detect and report memory leaks in real-time by just adding an attached property: https://github.com/AdamEssenmacher/MemoryToolkit.Maui

**Comment by PureWeen** [comment]
https://github.com/dotnet/maui/wiki/Memory-Leaks

**Comment by jurganson** [comment]
Just to reiterate what @PureWeen has mentioned - one example is that you can use the dotnet diagnostic tools to capture both CPU traces and gcdumps to profile your iOS and Android apps. For memory leaks you can capture gcdumps and view them through PerfView and get an overview of which .NET objects are holding your references and how many are leaking etc etc. The documentation linked has both iOS and Android guides, which (albeit long) are pretty straight forward to follow.
